native/source/mixpanel/detail/worker.cpp

```diff
@@ -135,7 +135,13 @@ namespace mixpanel
             nanowww::Response response;
             if (client.send_request(request, &response))
             {
-                mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "response: " + response.content());
+                // Lock on updating the network requests allowed time
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
```

Comment:
Great question!

Remember with C++ [RAII](https://en.wikipedia.org/wiki/Resource_Acquisition_Is_Initialization), so it handles the lock appropriately based on scope,